solidity/contracts/lib/zeto_nullifier.sol

```diff
@@ -82,11 +82,13 @@ abstract contract ZetoNullifier is IZeto, IZetoLockable, ZetoCommon {
             if (i > 0 && sortedOutputs[i] == sortedOutputs[i - 1]) {
                 revert UTXODuplicate(sortedOutputs[i]);
             }
-            bool existsInTree;
-            if (isLocked) {
-                (existsInTree, ) = existsAsLocked(sortedOutputs[i], msg.sender);
-            } else {
-                existsInTree = exists(sortedOutputs[i]);
+            // check the unlocked commitments tree
+            bool existsInTree = exists(sortedOutputs[i]);
```

Comment:
Can we renamed `exists` to `existsUnlocked` instead of adding a comment?

That could make bug like this one easier to spot when reading the code.